istioctl/pkg/validate/validate.go

```diff
@@ -110,10 +119,76 @@ func (v *validator) validateResource(un *unstructured.Unstructured) error {
 			Value: mixercrd.ToBackEndResource(un),
 		})
 	}
+	var errs error
+	if un.IsList() {
+		un.EachListItem(func(item runtime.Object) error { // nolint: errcheck
+			castItem := item.(*unstructured.Unstructured)
+			if castItem.GetKind() == "Service" {
+				err := v.validateServicePortPrefix(castItem)
+				if err != nil {
+					errs = multierror.Append(errs, err)
+				}
+			}
+			if castItem.GetKind() == "Deployment" {
+				v.validateDeploymentLabel(castItem)
+			}
+			return nil
+		})
+	}
 
+	if errs != nil {
+		return errs
+	}
+	if un.GetKind() == "Service" {
+		return v.validateServicePortPrefix(un)
+	}
+
+	if un.GetKind() == "Deployment" {
+		v.validateDeploymentLabel(un)
+		return nil
+	}
 	return nil
 }
 
+func (v *validator) validateServicePortPrefix(un *unstructured.Unstructured) error {
+	var errs error
+	if un.GetNamespace() == handleNamespace() {
+		return nil
+	}
+	spec := un.Object["spec"].(map[string]interface{})
+	if _, ok := spec["ports"]; ok {
+		ports := spec["ports"].([]interface{})
+		for _, port := range ports {
+			p := port.(map[string]interface{})
+			if p["protocol"] != nil && strings.EqualFold(p["protocol"].(string), serviceProtocolUDP) {
+				continue
+			}
+			if p["name"] == nil ||
+				servicePortPrefixed(p["name"].(string)) {
+				errs = multierror.Append(errs, fmt.Errorf("service %q is missing port name or the port name is incorrect", fmt.Sprintf("%s/%s/:",
```

Comment:
Can you break this into two messages?  Perhaps these messages:

"service %q has an unnamed port.  This is not recommended, see https://istio.io/docs/setup/kubernetes/prepare/requirements/"

and

"service %q port %q does not follow the Istio naming convention.  See https://istio.io/docs/setup/kubernetes/prepare/requirements/"